common/syscalls/syscalls.h

```diff
@@ -671,6 +671,10 @@ REV_SYSCALL( 9004, void dump_valid_mem( ) );
 REV_SYSCALL( 9005, void dump_valid_mem_to_file( const char* outputFile ) );
 REV_SYSCALL( 9006, void dump_thread_mem( ) );
 REV_SYSCALL( 9007, void dump_thread_mem_to_file( const char* outputFile ) );
+
+// ==================== REV PRINT UTILITIES
+REV_SYSCALL( 9110, int rev_fast_printf(const char* format, uint64_t a1=0, uint64_t a2=0, uint64_t a3=0, uint64_t a4=0, uint64_t a5=0, uint64_t a6=0) );
+
```

Comment:
Have you considered variadic templates (`uint64_t ...a`)?

They would allow an arbitrary number of arguments without having to use default values or placeholders.